### PR TITLE
Feat/sweep

### PR DESCRIPTION
**Pull request by TheLeoB**
## Description (optional)
Add a concise explanation of the changes.

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

**Review by Dezzj — COMMENTED** [review]
Veiwed

File: packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/sagas.ts [hunk 1]
```diff
@@ -566,8 +568,103 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
...
+    yield put(A.btcImportedFundsSweepLoading())
+    try {
+      // eslint-disable-next-line no-restricted-syntax
+      for (const addr of payload) {
+        const accounts = (yield select(selectors.core.common.btc.getAccountsBalances)).getOrElse([])
+        const defaultIndex = yield select(selectors.core.wallet.getDefaultAccountIndex)
+
+        const defaultAccount = accounts.filter((acc) => acc.index === defaultIndex)[0]
+
+        const receiveIndexMultiaddr = (yield select(
```
Comment (mrodriguez-bc) [inline packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/sagas.ts:614]: I dont see this as blocking the release, but one improvement here would have been to yield the initial index once, and then just increment on that. Instead, we are yielding a fetch to multiaddr each time.

File: packages/blockchain-wallet-v4/src/redux/payment/bch/sagas.js [hunk 2]
```diff
@@ -48,12 +48,12 @@ export default ({ api }) => {
       .then(map(toCoin(network, fromData)))
 
   // ///////////////////////////////////////////////////////////////////////////
-  const calculateTo = function* (destinations, type, network) {
+  const calculateTo = function* (destinations, type, addressIndex, network) {
```
Comment (mrodriguez-bc) [inline packages/blockchain-wallet-v4/src/redux/payment/bch/sagas.js:51]: this was clever

**Review by mrodriguez-bc — APPROVED** [review]

File: packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/sagas.ts [hunk 3]
```diff
@@ -566,8 +568,103 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
...
 
+  const btcImportedFundsSweepEffectiveBalance = function* () {
+    try {
+      yield take(actionTypes.core.data.btc.FETCH_BTC_DATA_SUCCESS)
+      const addressesR = yield select(selectors.core.common.btc.getActiveAddresses)
+      const addresses = addressesR.getOrElse([]).filter(prop('priv'))
+      const hasEffectiveBalances: ImportedBtcAddressList = []
+      // eslint-disable-next-line no-restricted-syntax
+      for (const addr of addresses) {
+        if (addr.info.final_balance > 0) {
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/sagas.ts:579]: is there a chance that addr.info is undefined?

**Review by milan-bc — APPROVED** [review]

File: packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/sagas.ts (hunk 3 above)
Comment (TheLeoB) [reply packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/sagas.ts:579]: if addresses is an empty array, it jumps out of the loop so it doesn't run the next line. In theory no, I tested it with wallets without imported addresses. I'll handle that case in next PR.

**Pull request merged**